torchao/_models/mixtral-moe/generate.py

```diff
@@ -0,0 +1,393 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+import itertools
```

Comment:
Do we need all this? 

I understand it is good to have an MOE model but is it important to have our own definition here?

Reply:
we've discussed adding a moe model for a while and this was the main one for testing and benchmarking and demonstrating the technique.